### PR TITLE
Adding posts count to dashboard

### LINKED ISSUE #2419: Show Posts count on status dashboard
In #2395 we are adding an expanded status service dashboard. One thing I want to include in it is the ability to see the current number of Posts and Feeds in the database.  We expose this data as a header when you query for `/posts` or `/feeds` from the Posts service:

https://github.com/Seneca-CDOT/telescope/blob/master/src/api/posts/src/routes/posts.js#L55

Let's add a widget to the dashboard that shows the current counts for each of these.

### PR DESCRIPTION
**Pull request by nguyenhung15913**
## Issue This PR Addresses

Fixes #2419 
## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description

- Added <code>cors</code> to Satellite options at <code>src/api/posts/src/index.js</code> to expose <code>x-total-count</code> header.
- Added a component to show posts count at <code>src/api/status/public/index.html</code>
- Added <code>postCounts.js</code> at <code>src/api/status/public</code> to fetch and get the total of posts
- Added host API to <code>connectSrc</code> of Satellite options, and <code>cors</code> at <code>src/api/status/src/server.js</code> 

Note: I worked with <code>postsCount.js</code> locally using fetch to get the <code>x-total-count</code> header from <code>https://localhost/v1/posts</code> and it worked. However, with <code>https://dev.api.telescope.cdot.systems/v1/posts</code>, <code>response.headers.get(x-total-count)</code> will return null.

To test locally:
- In <code>src/api/status/env.local</code>, include <code>API_HOST=http://localhost/v1/posts</code>
- Comment on ```<base href="/v1/status/" />``` in <code>src/api/status/public/index.html</code> (This line however is required in production)
- add <code>process.env.API_HOST</code> to <code>connectSrc</code> in <code>src/api/status/src/server.js</code>
- Change fetch <code>url</code> to 'http://localhost/v1/posts' instead in <code>src/api/status/public/js/post/index.js</code>
- Run posts service in the root directory <code>npm run services:start posts</code>
- Run <code>npm run dev</code> in <code>src/api/status</code>
Here is the screenshot of working posts count:
image

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

<a href="https://gitpod.io/#https://github.com/Seneca-CDOT/telescope/pull/2476"></a>

### CODE REVIEW

File: src/api/status/src/server.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+var app = express();
 
+app.use(cors({ origin: '*' }));
 const satelliteOptions = {
   helmet: {
     contentSecurityPolicy: {
       directives: {
         defaultSrc: ["'self'"],
         scriptSrc: ["'self'", '*.fontawesome.com'],
-        styleSrc: ["'self'", "'unsafe-inline'", '*.fontawesome.com', 'fonts.googleapis.com'],
```
Comment (menghif) [inline src/api/status/src/server.js:11]: 'fonts.googleapis.com' is needed for the icons

**Review by menghif — COMMENTED** [review]
I'm getting Total Posts:  N/A when running this. Is there something I need to change before running it?

File: src/api/status/public/index.html [hunk 2]
```diff
@@ -16,7 +16,6 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <base href="/v1/status/" />
```
Comment (Andrewnt219) [inline src/api/status/public/index.html:19]: This is necessary for deployment, only remove it in local dev.

File: src/api/status/public/index.html [hunk 3]
```diff
@@ -1224,5 +1244,6 @@ <h6 class="mt-3">Thank you for sharing!</h6>
     <script src="assets/js/demo-sidenav.js"></script>
     <!-- Control Center for Material Dashboard: parallax effects, scripts for the example pages etc -->
     <script src="assets/js/material-dashboard.min.js?v=3.0.0"></script>
+    <script src="postsCount.js"></script>
```
Comment (Andrewnt219) [inline src/api/status/public/index.html:1247]: Branch is outdated. There should be `/public/js` for js files.

File: src/api/status/public/index.html [hunk 4]
```diff
@@ -410,6 +409,27 @@ <h4 class="mb-0">$103,430</h4>
               </div>
             </div>
           </div>
+          <div class="col-xl-3 col-sm-6 mb-xl-0 mb-4 mt-4">
```
Comment (Andrewnt219) [inline src/api/status/public/index.html:412]: Think this should wait for @menghif  PR to have the same style?

File: src/api/status/public/postsCount.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const totalPosts = document.getElementById('totalPosts');
```
Comment (Andrewnt219) [inline src/api/status/public/postsCount.js:1]: Just nitpicking, but `totalPostsEl` can be more clear here.

File: src/api/status/public/postsCount.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+const fetchPostsCount = () =>
+  fetch('dev.api.telescope.cdot.systems').then((res) => res.headers.get('x-total-count'));
+
+const getPostsCount = async () => {
+  try {
+    const total = await fetchPostsCount();
+    totalPosts.innerText = total || 'N/A';
```
Comment (Andrewnt219) [inline src/api/status/public/postsCount.js:8]: This should use [null-coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) `total ?? "N/A"`, as 0 is a falsy value.

File: src/api/status/src/server.js [hunk 7]
```diff
@@ -1,15 +1,19 @@
 const { Satellite } = require('@senecacdot/satellite');
 const { static } = require('express');
 const path = require('path');
+var express = require('express');
+var cors = require('cors');
+var app = express();
 
+app.use(cors({ origin: '*' }));
```
Comment (Andrewnt219) [inline src/api/status/src/server.js:8]: This probably has no effects. The server is started by Satellite. Please revert the changes.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

File: src/api/status/env.local [hunk 8]
```diff
@@ -1 +1,2 @@
 PLANET_PORT=1111
+API_URL=http://localhost/v1/posts
```
Comment (Andrewnt219) [inline src/api/status/env.local:2]: This is not necessary, as mentioned [here](https://github.com/Seneca-CDOT/telescope/pull/2471#discussion_r749590703)

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

File: src/api/status/env.local (hunk 8 above)
Comment (tuanthanh2067) [reply src/api/status/env.local:2]: Actually we had to add cors to posts api and run it locally so that we could retrieve x-total-count from the header. That's why he put `localhost/v1/posts` there.

Comment (Andrewnt219) [reply src/api/status/env.local:2]: but it shouldn't be included in the commit right? And for cors, it is passed inside the Satellite as an option, not a separate express app. 
https://github.com/Seneca-CDOT/telescope/blob/1407a343df43b0c2d2bac941e4351af1ced68a43/src/api/posts/src/index.js#L7-L10

Comment (tuanthanh2067) [reply src/api/status/env.local:2]: Yeah it shouldn't be in `env.local`

**Comment by nguyenhung15913** [comment]
> I'm getting Total Posts: N/A when running this. Is there something I need to change before running it?

Hi Mengi, you should change the URL in the <code>fetch</code> in <code>postsCount.js</code> to <code>https://localhost/v1/posts</code>. run <code>npm run services:start posts</code> in the root folder. This will resolve the issue. This is because as I mentioned, It only worked when you work locally.

File: src/api/status/public/index.html (hunk 4 above)
Comment (nguyenhung15913) [reply src/api/status/public/index.html:412]: For sure. I just copy paste the same component of the page for testing posts count.

File: src/api/status/public/index.html (hunk 3 above)
Comment (tuanthanh2067) [reply src/api/status/public/index.html:1247]: Yeah I think we have `index.js` under /public/js/feed

Comment (nguyenhung15913) [reply src/api/status/public/index.html:1248]: Hi, how can I fix this issue? I don't really understand it

Comment (tuanthanh2067) [reply src/api/status/public/index.html:1248]: What I would do is that I will include `script.js` in the `index.html` and have it call the `setPostCount` or whatever it is and `setPostCount` should be placed in `index.js` which is under `/public/js/post`

File: src/api/status/public/index.html [hunk 9]
```diff
@@ -410,6 +410,27 @@ <h4 class="mb-0">$103,430</h4>
...
+                  <i class="material-icons opacity-10">person</i>
+                </div>
+                <div class="text-end pt-1">
+                  <p class="text-sm mb-0 text-capitalize">Total posts</p>
+                  <h4 class="mb-0" id="totalPosts"></h4>
+                </div>
+              </div>
+              <hr class="dark horizontal my-0" />
+              <div class="card-footer p-3">
+                <p class="mb-0">
```
Comment (menghif) [inline src/api/status/public/index.html:428]: The `<p>` "-2% than yesterday" should be removed

**Review by menghif — CHANGES_REQUESTED** [review]
File name should be changed to: `posts-count.js`

File: src/api/status/public/index.html (hunk 3 above)
Comment (Andrewnt219) [reply src/api/status/public/index.html:1248]: @nguyenhung15913 The current master branch has a directory for js files, you should rebase your beach.

image

**Comment by humphd** [comment]
This needs a rebase

File: src/api/status/public/index.html (hunk 4 above)
Comment (Andrewnt219) [reply src/api/status/public/index.html:477]: Can you match this card with all the previous cards? And if you don't have a footer, use `card-body` instead of `card-header`. Your current card is losing the rounded bottom corners.

File: src/api/status/public/js/posts-count.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
+const totalPostsEl = document.getElementById('totalPosts');
+const fetchPostsCount = () =>
+  fetch('https://dev.api.telescope.cdot.systems/v1/posts').then((res) =>
```
Comment (Andrewnt219) [inline src/api/status/public/js/posts-count.js:3]: This should be using `process.env.API_HOST` instead. In production, `process.env.API_HOST` is  `dev.api.telescope.cdot.systems`

https://github.com/Seneca-CDOT/telescope/blob/master/config/env.production#L21

Comment (Andrewnt219) [reply src/api/status/public/js/posts-count.js:3]: Also, you `fetch(url, {method: 'HEAD'})`

https://github.com/Seneca-CDOT/telescope/pull/2471/files#r752232958

File: src/api/status/public/js/posts-count.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+const getPostsCount = async () => {
+  try {
+    const total = await fetchPostsCount();
+    totalPostsEl.innerText = total ?? 'N/A';
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+window.onload = function () {
```
Comment (Andrewnt219) [inline src/api/status/public/js/posts-count.js:16]: In public/script.js, `window.onload` is already assigned. Call your function in there.

https://github.com/Seneca-CDOT/telescope/blob/00213c8cd1e9020196dcc892772fc04f4f0fdc23/src/api/status/public/script.js#L51-L53

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]
Can you include a screenshot of your work in the PR?

File: src/api/status/public/index.html [hunk 12]
```diff
@@ -474,6 +474,22 @@ <h4 class="mb-0" id="total-contributors-satellite">&nbsp;</h4>
...
+                  class="icon icon-lg icon-shape bg-gradient-success shadow-success text-center border-radius-xl mt-n4 position-absolute"
+                >
+                  <i class="material-icons opacity-10">person</i>
+                </div>
+                <div class="text-end pt-1">
+                  <p class="text-sm mb-0 text-capitalize">Total posts</p>
+                  <h4 class="mb-0" id="totalPosts"></h4>
+                </div>
+              </div>
+              <hr class="dark horizontal my-0" />
```
Comment (menghif) [inline src/api/status/public/index.html:490]: This is breaking the round design:
Screen Shot 2021-11-18 at 3 34 49 PM

You can remove this `<hr>` and add a `<div>` with `class="card-footer"`. Something like this should fix it.

**Review by menghif — CHANGES_REQUESTED** [review]

File: src/api/status/public/js/posts-count.js (hunk 10 above)
Comment (nguyenhung15913) [reply src/api/status/public/js/posts-count.js:3]: > process.env.API_HOST

will it work? As I know, the browser cannot read environment variable.

Comment (humphd) [reply src/api/status/public/js/posts-count.js:3]: Correct, we can't read `process` in the browser.

What we need here is a function to get the API Host for us, something like:

```js
const apiHost = new URL(location).origin;

...

fetch(`${apiHost}/v1/posts`)
```

File: src/api/status/public/js/posts-count.js (hunk 11 above)
Comment (tuanthanh2067) [reply src/api/status/public/js/posts-count.js:16]: I don't think we need to call it on `window.onload` instead we can call it in its separate file and have the file included in the `index.html`

Comment (humphd) [reply src/api/status/public/js/posts-count.js:16]: Either way it needs to wait on the DOM being ready.

File: src/api/status/public/js/posts-count.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+const totalPostsEl = document.getElementById('totalPosts');
+const fetchPostsCount = () =>
+  fetch('https://dev.api.telescope.cdot.systems/v1/posts').then((res) =>
+    res.headers.get('x-total-count')
+  );
+
+const getPostsCount = async () => {
+  try {
+    const total = await fetchPostsCount();
+    totalPostsEl.innerText = total ?? 'N/A';
```
Comment (tuanthanh2067) [inline src/api/status/public/js/posts-count.js:10]: We should leave innerText blank or N/A because I need to know to make mine the same.

Comment (nguyenhung15913) [reply src/api/status/public/js/posts-count.js:10]: > We should leave innerText blank or N/A because I need to know to make mine the same.

I think it should be N/A to make it clearer

File: src/api/posts/package.json [hunk 14]
```diff
@@ -17,6 +17,7 @@
   },
   "dependencies": {
     "@senecacdot/satellite": "^1.x",
+    "dotenv": "^10.0.0",
```
Comment (humphd) [inline src/api/posts/package.json:20]: This change doesn't seem necessary

File: src/api/status/env.local [hunk 15]
```diff
@@ -1 +1,3 @@
 PLANET_PORT=1111
+
```
Comment (humphd) [inline src/api/status/env.local:2]: This isn't necessary either.

File: src/api/status/public/index.html (hunk 4 above)
Comment (humphd) [reply src/api/status/public/index.html:477]: This needs to get addressed per @Andrewnt219's comment.

File: src/api/status/public/js/post/index.js [hunk 16]
```diff
@@ -0,0 +1,19 @@
+const totalPostsEl = document.getElementById('totalPosts');
```
Comment (humphd) [inline src/api/status/public/js/post/index.js:1]: This is dangerous to do outside of onload, since the DOM may or may not be ready, and you'll get race conditions.

Comment (humphd) [reply src/api/status/public/js/post/index.js:1]: Move it to line 11, when you use it.

File: src/api/status/public/script.js [hunk 17]
```diff
@@ -50,4 +50,5 @@ function checkStatus() {
 
 window.onload = function () {
   checkStatus();
+  getPostsCount();
```
Comment (humphd) [inline src/api/status/public/script.js:53]: This isn't necessary to do twice.

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/api/status/public/js/post/index.js [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+
+const apiHost = new URL(location).origin;
+
+const fetchPostsCount = () =>
+  fetch(`${apiHost}/v1/posts`, { method: 'HEAD' }).then((res) => res.headers.get('x-total-count'));
```
Comment (tuanthanh2067) [inline src/api/status/public/js/post/index.js]: In dev mod, the path `https://dev.api.telescope.cdot.systems/v1/posts` isn't avaiable and the base url is `https://dev.api.telescope.cdot.systems/v1/status`, so we should replace status with `posts`

File: src/api/status/public/js/post/index.js [hunk 19]
```diff
@@ -0,0 +1,19 @@
+const totalPostsEl = document.getElementById('totalPosts');
+
+const apiHost = new URL(location).origin.replace(/status/, '');
```
Comment (tuanthanh2067) [inline src/api/status/public/js/post/index.js:3]: I don't think this is necessary as origin will only return back `https://dev.api.telescope.cdot.systems/`

File: src/api/status/public/js/post/index.js [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+
+const fetchPostsCount = () =>
+  fetch(`${apiHost}posts`, { method: 'HEAD' }).then((res) => res.headers.get('x-total-count'));
```
Comment (tuanthanh2067) [inline src/api/status/public/js/post/index.js:6]: this should be `${apiHost}/v1/posts`

Comment (nguyenhung15913) [reply src/api/status/public/js/post/index.js]: fixed

File: src/api/status/public/index.html (hunk 4 above)
Comment (nguyenhung15913) [reply src/api/status/public/index.html:477]: @Andrewnt219 So i will need to wait for @menghif PR to have the same style?

File: src/api/status/public/index.html (hunk 12 above)
Comment (nguyenhung15913) [reply src/api/status/public/index.html]: Hi I fixed. Can you check if its the same as your design ?

**Comment by humphd** [comment]
We just merged @tuanthanh2067's work, which overlaps with this.  Please rebase to pick it up.

File: src/api/status/public/js/post/index.js [hunk 21]
```diff
@@ -0,0 +1,18 @@
+const apiHost = new URL(location).origin;
+
+const fetchPostsCount = () =>
+  fetch(`${apiHost}/v1/posts`, { method: 'HEAD' }).then((res) => res.headers.get('x-total-count'));
+
+const getPostsCount = async () => {
+  try {
+    const total = await fetchPostsCount();
```
Comment (trangntt-016) [inline src/api/status/public/js/post/index.js:8]: I think it's better to change "total" to "numberOfPosts" which makes it much more readable

**Review by trangntt-016 — CHANGES_REQUESTED** [review]

File: src/api/status/public/index.html (hunk 4 above)
Comment (Andrewnt219) [reply src/api/status/public/index.html:477]: It was merged 2 days ago. The PR is already on your branch. Just need to copy the structure from the card with `card--telescope` on it.

https://github.com/Seneca-CDOT/telescope/blob/7368bcd49ed6cca1e36a3278c317a8c95f0d251e/src/api/status/public/index.html#L332

File: src/api/status/env.local [hunk 22]
```diff
@@ -1 +0,0 @@
-PLANET_PORT=1111
```
Comment (Andrewnt219) [inline src/api/status/env.local:1]: why is this line removed?

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]
There're still conflicts with the master branch.

File: src/api/status/public/js/post/index.js [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+    const totalPostsEl = document.getElementById('totalPosts');
+    totalPostsEl.innerText = total ?? 'N/A';
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+window.onload = function () {
+  getPostsCount();
```
[the remark this answers is not in the document]
Comment (nguyenhung15913) [reply src/api/status/public/js/post/index.js:17]: Hi, I created a js/post/index.js and included it to index.html . I want to use the same structure as the Feeds Count of @tuanthanh2067. If you still want the change, I can do it too.

File: src/api/status/env.local (hunk 22 above)
Comment (nguyenhung15913) [reply src/api/status/env.local:1]: Hi, I got a comment here from @humphd. I think this is not necessary.
https://github.com/Seneca-CDOT/telescope/pull/2476#discussion_r752791610

**Comment by nguyenhung15913** [comment]
> Almost there, just some small changes. There're still conflicts with the master branch.
> 
> EDIT: Can you also update the PR description with some screenshots, and how to test your function in local?

I am  doing it now.

File: src/api/status/env.local (hunk 22 above)
Comment (tuanthanh2067) [reply src/api/status/env.local:1]: That's actually a new line unnecessary not the `PLANET_PORT`

Comment (nguyenhung15913) [reply src/api/status/env.local:1]: I just realized that, Added back the <code>env.local</code>

File: src/api/status/public/js/post/index.js (hunk 21 above)
Comment (nguyenhung15913) [reply src/api/status/public/js/post/index.js:8]: Makes sense. I will change it

File: src/api/status/public/index.html (hunk 4 above)
Comment (humphd) [reply src/api/status/public/index.html:477]: This seems not done. @nguyenhung15913 can you please do it and then resolve this conversation?

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/api/status/public/js/post/index.js (hunk 23 above)
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply src/api/status/public/js/post/index.js:17]: That's the flaw with the current structure. Now we have two assignments to `window.onload`. This means one will override the other.

[the remark this answers is not in the document]
Comment (humphd) [reply src/api/status/public/js/post/index.js:17]: As an aside, this is the point of `addEventListener()` vs. `onload`: it lets you have as many listeners registered as you need.

[the remark this answers is not in the document]
Comment (tuanthanh2067) [reply src/api/status/public/js/post/index.js:17]: Oh that's the problem, so I think we should change it to `addEventListener()` instead of `window.onload`

**Comment by nguyenhung15913** [comment]
@Andrewnt219 I created an <code>index.js</code> in <code>js</code> folder, included it in <code>index.html</code> and it imported and called <code>getPostsCount()</code>

File: src/api/status/public/js/post/index.js (hunk 23 above)
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply src/api/status/public/js/post/index.js]: @humphd  @tuanthanh2067  Think it's easier to track what is called on the page load when all the calls are in one place, and `addEvenListener` as a fallback so we don't mess up any other scripts (e.g. third-party scripts).

**Comment by Andrewnt219** [comment]
Nice, can you rebase your branch, and align the cards of feed and post on the same row?

File: src/api/status/public/js/post/index.js (hunk 23 above)
[the remark this answers is not in the document]
Comment (tuanthanh2067) [reply src/api/status/public/js/post/index.js]: Yeah but in [here](https://github.com/Seneca-CDOT/telescope/commit/3131812c364e551b697c123aa9a66df695b9e723#diff-3ac2b0a47760bda0ee48fbaf19723362505a75f7d5b0eb955dc72dbf7f8ac41eR64-R66). @menghif also have it called in its own file

File: src/api/status/public/index.html (hunk 4 above)
Comment (Andrewnt219) [reply src/api/status/public/index.html:477]: This need to be put inside a row, together with the feed card so they can look like the first 4 cards. Add the margin-top to the row container, not the cards.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

**Comment by tuanthanh2067** [comment]
I just realized the card for feed is not quite the size of other cards. I'm wondering if I can sneak it to this pull request.

File: src/api/status/public/js/post/index.js (hunk 23 above)
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply src/api/status/public/js/post/index.js]: that's why I want to fix the current structure, but we can do it in a follow-up.

**Comment by Andrewnt219** [comment]
The branch still need to rebase
image

[This request](https://github.com/Seneca-CDOT/telescope/pull/2476#discussion_r753523895) is not resolved

**Comment by menghif** [comment]
Squash and rebase and this is good to go!

**Comment by nguyenhung15913** [comment]
> The branch still need to rebase image
> 
> [This request](https://github.com/Seneca-CDOT/telescope/pull/2476#discussion_r753523895) is not resolved

I fixed that already, here is the result:
image

**Comment by nguyenhung15913** [comment]
Hi @Andrewnt219 , When i do <code>git rebase master</code>, I have this issue:
image
Then if I <code>git rebase --continue or git rebase --skip</code>, I will get this:
image
Do you know what is the issue ?

**Comment by Andrewnt219** [comment]
seem like you're in the middle of another rebase. Abort the previous rebase with `git rebase --abort`. **Any resolved conflicts during that previous rebase is also lost**.

**Comment by menghif** [comment]
@nguyenhung15913 Also revert this: "Merge branch 'master' into postsCount"
You shouldn't merge master in your branch before doing a rebase

**Comment by Andrewnt219** [comment]
You messed up something during the rebase, changes from other PRs now show up in your branch.

File: config/env.development [hunk 24]
```diff
@@ -18,7 +18,7 @@ TELESCOPE_HOST=http://localhost:3000
 
 # The host where all the microservices run (e.g., http://localhost)
 # NOTE: if you change this, change all other occurrences below too.
-API_HOST=localhost
+API_HOST=http://localhost
```
Comment (Andrewnt219) [inline config/env.development:21]: From a different PR

File: docker/docker-compose.yml [hunk 25]
```diff
@@ -295,7 +295,7 @@ services:
 
   # API Gateway
   traefik:
-    image: traefik:v2.5.0
+    image: traefik:v2.4.5
```
Comment (Andrewnt219) [inline docker/docker-compose.yml:298]: From a different PR

File: package.json [hunk 26]
```diff
@@ -82,8 +82,8 @@
     "opml-generator": "1.1.1",
     "passport": "0.4.1",
     "passport-saml": "3.1.2",
-    "pino": "7.3.0",
-    "pino-elasticsearch": "6.2.0",
+    "pino": "6.11.2",
```
Comment (Andrewnt219) [inline package.json:85]: From a different PR

File: src/api/posts/package.json (hunk 14 above)
Comment (Andrewnt219) [reply src/api/posts/package.json:20]: Also need to revert this.

File: src/api/status/src/server.js [hunk 27]
```diff
@@ -19,7 +18,7 @@ const satelliteOptions = {
         connectSrc: [
           "'self'",
           '*.fontawesome.com',
-          `${host.replace(/(^\w+:|^)\/\//, '')}:4000`,
+          `${process.env.API_HOST.replace(/(^\w+:|^)\/\//, '')}:4000`,
```
Comment (Andrewnt219) [inline src/api/status/src/server.js:21]: Revert changes to this whole file.

File: src/api/status/src/server.js [hunk 28]
```diff
@@ -31,16 +30,8 @@ const satelliteOptions = {
...
-if (process.env.PATH_PREFIX) {
-  service.router.use(process.env.PATH_PREFIX, static(path.join(__dirname, '../public')));
-  service.router.get('/', (req, res) => {
-    res.redirect(process.env.PATH_PREFIX);
-  });
-} else {
-  service.router.use('/', static(path.join(__dirname, '../public')));
-}
+// Static web assets can be cached for a long time
+service.router.use('/', static(path.join(__dirname, '../public')));
```
Comment (Andrewnt219) [inline src/api/status/src/server.js:34]: Revert changes to this whole file.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

**Comment by nguyenhung15913** [comment]
> You messed up something during the rebase, changes from other PRs now show up in your branch.

what I did is fetch the upstream master to get the all the work. Then I <code>git pull</code> on my master branch, then I <code>git rebase master</code> from my <code>postsCount</code> branch

**Comment by Andrewnt219** [comment]
Sounds right, perhaps the errors are during the conflict solving process (i.e. currrent/incoming changes). Current is the master changes, incoming is your changes.

**Comment by nguyenhung15913** [comment]
> Sounds right, perhaps the errors are during the conflict solving process (i.e. currrent/incoming changes). Current is the master changes, incoming is your changes.

You have any idea about solving this issue, or i just have to rebase again ?

File: package.json (hunk 26 above)
Comment (humphd) [reply package.json:85]: `git checkout master package.json`

File: src/api/posts/package.json (hunk 14 above)
Comment (humphd) [reply src/api/posts/package.json:20]: `git checkout master src/api/posts/package.json`

File: src/api/status/env.local [hunk 29]
```diff
@@ -1,3 +1 @@
 PLANET_PORT=1111
-PATH_PREFIX=/v1/status
-API_HOST=localhost
```
Comment (humphd) [inline src/api/status/env.local:3]: Don't touch this file

```
git checkout master src/api/status/env.local
```

File: src/api/status/public/index.html [hunk 30]
```diff
@@ -1307,5 +1328,6 @@ <h6 class="mt-3">Thank you for sharing!</h6>
     <!-- Telescope JS Files -->
     <script src="js/github-stats.js"></script>
     <script src="js/feed/index.js"></script>
+    <script type="module" src="js/index.js"></script>
```
Comment (humphd) [inline src/api/status/public/index.html:1331]: Can we please follow the same pattern as these other two files for naming and type?

File: src/api/status/src/server.js [hunk 31]
```diff
@@ -1,7 +1,6 @@
 const { Satellite } = require('@senecacdot/satellite');
```
Comment (humphd) [inline src/api/status/src/server.js:1]: Revert this too:

```
git checkout master src/api/status/src/server.js
```

When you rebase later, this has all been fixed.

**Review by humphd — CHANGES_REQUESTED** [review]

**Comment by nguyenhung15913** [comment]
@Andrewnt219 @humphd I tried to fetch upstream in my current forked repo. But seems like some of the files could not update. One of them is docker/docker-compose.yml (<code>image: traefik:v2.4.5</code> instead of <code>image: traefik:v2.5.0</code>). Should I re-fork and make another pull request?

**Comment by humphd** [comment]
You need to revert the changes I mentioned, so that it can pull in the proper ones when you rebase onto master.

File: src/api/status/public/index.html (hunk 30 above)
Comment (Andrewnt219) [reply src/api/status/public/index.html:1159]: @nguyenhung15913 this should be changed to js/pages/dashboard.js

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]
The rebase is correct now, I tested on local but the "http://localhost/v1/posts" return an empty array. Other than that, everything is great.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]
Good to go, just revert the unnecessary change and squash all the commits.

File: src/api/status/public/pages/build.html [hunk 32]
```diff
@@ -2,7 +2,7 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <base href="/v1/status/" />
+    <!-- <base href="/v1/status/" /> -->
```
Comment (Andrewnt219) [inline src/api/status/public/pages/build.html:5]: Revert this. Running `yarn dev` inside /src/ap/status should be working out-of-the-box.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

File: tatus [hunk 33]
```diff
@@ -0,0 +1,19124 @@
+[33mcommit ca960ffaf56c090ec63a5d0a823b50c6e374bc84[m[33m ([m[1;36mHEAD -> [m[1;32mmaster[m[33m, [m[1;31morigin/master[m[33m, [m[1;31morigin/HEAD[m[33m)[m
```
Comment (Andrewnt219) [inline tatus:1]: What is this file? Should remove this.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

File: config/env.development (hunk 24 above)
Comment (Andrewnt219) [reply config/env.development:21]: Revert.

File: docker/docker-compose.yml (hunk 25 above)
Comment (Andrewnt219) [reply docker/docker-compose.yml:298]: Revert.

File: package.json (hunk 26 above)
Comment (Andrewnt219) [reply package.json:85]: Revert.

File: src/api/status/env.local [hunk 34]
```diff
@@ -1,3 +1 @@
 PLANET_PORT=1111
```
Comment (Andrewnt219) [inline src/api/status/env.local:1]: Revert.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]
The rebase went wrong again...

File: src/api/status/public/js/pages/dashboard.js [hunk 35]
```diff
@@ -0,0 +1,10 @@
+import { getPostsCount } from '../post/index';
```
Comment (Andrewnt219) [inline src/api/status/public/js/pages/dashboard.js:1]: These 3 imports are missing `.js` extension at the end.

File: src/api/status/src/server.js [hunk 36]
```diff
@@ -2,7 +2,6 @@ const { Satellite } = require('@senecacdot/satellite');
 const { static } = require('express');
 const path = require('path');
 const host = process.env.API_HOST || 'localhost';
-
```
Comment (Andrewnt219) [inline src/api/status/src/server.js:5]: Revert this unnecessary change.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

**Review by Andrewnt219 — APPROVED** [review]
Great work! 

I tested locally and this should be ready to merge @humphd.

File: src/api/status/public/js/pages/dashboard.js [hunk 37]
```diff
@@ -0,0 +1,10 @@
+import { getPostsCount } from '../post/index.js';
+import { getFeedCount } from '../feed/index.js';
+import { getGitHubData } from '../github-stats.js';
+
+window.addEventListener('load', (event) => {
```
Comment (humphd) [inline src/api/status/public/js/pages/dashboard.js:5]: Minor nit: `event` isn't used, you should remove it.

**Review by humphd — APPROVED** [review]
